app/models/crm/teaching_subject.rb

```diff
@@ -51,27 +23,35 @@ def lookup_by_keys(*keys)
     end
 
     def lookup_by_uuid(uuid)
-      ALL.invert[uuid]
+      all_hash.invert[uuid]
     end
 
     def key_with_uuid(uuid)
       keyed_subjects.invert[uuid]
     end
 
     def all_uuids
-      ALL.values
+      all_hash.values
     end
 
     def all_subjects
-      ALL.keys
+      all_hash.keys
     end
 
-    def ignore?(uuid)
-      IGNORED.value?(uuid)
+    def keyed_subjects
+      @keyed_subjects ||= all_hash.transform_keys { |key| key.parameterize(separator: "_").to_sym }
     end
 
-    def keyed_subjects
-      @keyed_subjects ||= ALL.transform_keys { |key| key.parameterize(separator: "_").to_sym }
+    def all_hash
+      all.to_h { |item| [item.value, item.id] }
+    end
+
+    def all_without_primary
+      all_hash.reject { |_h, s| s == PRIMARY }
+    end
+
+    def all
+      GetIntoTeachingApiClient::LookupItemsApi.new.get_teaching_subjects.reject { |subject| IGNORED.value?(subject.id) }.each { |subject| if subject && RENAMED[subject.value].present? then subject.value = RENAMED[subject.value]; end; }.sort_by { |f| [f.value] }
```

Comment:
I think this needs to be simplified its quite hard to follow. We could condense it to something along the lines of:

```
def all
  GetIntoTeachingApiClient::LookupItemsApi.new.get_teaching_subjects
    .sort_by(&:value)
    .reject(&method(:ignored?))
    .each(&method(:rename))
end

private

def rename(subject)
  subject.value = RENAMED[subject.value] || subject.value
end

def ignored?(subject)
  IGNORED.value?(subject.id)
end
```

As a follow-up PR I think we should look at removing the `all_hash` from this class, but that would have many knock-on changes so it can be done separately

Reply:
yeah, good idea, I just didn't look through as just fixed it so rubocop doesn't complain